clang/lib/CodeGen/CGHLSLBuiltins.cpp

```diff
@@ -299,8 +299,18 @@ Value *CodeGenFunction::EmitHLSLBuiltinExpr(unsigned BuiltinID,
     // (llvm/llvm-project#135452)
     Value *NonUniform =
         llvm::ConstantInt::get(llvm::Type::getInt1Ty(getLLVMContext()), false);
+
+    auto [IntrinsicID, HasNameArg] =
+        CGM.getHLSLRuntime().getCreateHandleFromBindingIntrinsic();
+    if (HasNameArg) {
+      Value *NameOp = EmitScalarExpr(E->getArg(5));
+      return Builder.CreateIntrinsic(HandleTy, IntrinsicID,
+                                     ArrayRef<Value *>{SpaceOp, RegisterOp,
+                                                       RangeOp, IndexOp,
+                                                       NonUniform, NameOp});
```

Comment:
It might simplify the code slightly to do something like
```c++
    SmallVector<Value *> Args{SpaceOp, RegisterOp, RangeOp, IndexOp, NonUniform};
    if (HasNameArg)
      Args.append(EmitScalarExpr(E->getArg(5)));
```